src/box/lua/config/source/env.lua

```diff
@@ -14,6 +15,62 @@ function methods._env_var_name(self, path_in_schema)
     return env_var_name
 end
 
+function methods._env_list(self)
+    local rows = {}
+
+    local ce = 'Community Edition'
+    local ee = 'Enterprise Edition'
+
+    -- A header of the table.
+    table.insert(rows, {
+        'ENVIRONMENT VARIABLE',
+        'TYPE',
+        'DEFAULT',
+        'AVAILABILITY',
+    })
+    table.insert(rows, tabulate.SPACER)
+
+    -- Environment variables that duplicate CLI options.
+    table.insert(rows, {
+        'TT_INSTANCE_NAME',
+        'string',
+        'N/A',
+        ce,
+    })
+    table.insert(rows, {
+        'TT_CONFIG',
+        'string',
+        'nil',
+        ce,
+    })
+    table.insert(rows, tabulate.SPACER)
+
+    -- Collect the options from the schema and sort them
+    -- lexicographically.
+    local options = instance_config:pairs():totable()
+    table.sort(options, function(a, b)
+        return table.concat(a.path, '_') < table.concat(b.path, '_')
+    end)
+
+    -- Transform the schema nodes (options) into the environment
+    -- variables description.
+    for _, w in ipairs(options) do
+        local default = w.schema.default
+        if default == nil and type(default) == 'cdata' then
+            default = 'box.NULL'
+        end
+
+        table.insert(rows, {
+            self._env_var_name(self, w.path),
+            w.schema.type,
+            tostring(default),
+            w.schema.computed.annotations.enterprise_edition and ee or ce,
```

Comment:
Added a short note into the commit message of the first commit.

> It will be used in one of the next commits this way:
>
> ```lua
> for _, w in instance_config:pairs() do
>     if w.schema.computed.annotations.enterprise_edition then
>         <...>
>     else
>         <...>
>     end 
> end
> ```